pika/adapters/base_connection.py

```diff
@@ -239,7 +241,10 @@ def _get_error_code(self, error_value):
         return None
 
     def _flush_outbound(self):
-        """Call the state manager who will figure out that we need to write."""
+        """write early, if the socket will take the data why not get it out
+        there asap.
+        """
+        self._handle_write()
```

Comment:
Adding a call to `_handle_write()` here shouldn't be necessary with the Twisted framework: if there is anything in `outbound_buffer`, then `_manage_event_state()` will call `IOLoopReactorAdapter.update_handler()` with the `WRITE` bit turned on, resulting in a call to `self.reactor.addWriter(self.connection)`, and a subsequent call in short order to `TwistedConnection.doWrite()` from the reactor if/when the socket is writable. I think that it would be cleaner to just let the Twisted reactor do its job naturally.